contracts/registries/LicenseRegistryV2.sol

```diff
@@ -0,0 +1,382 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.23;
+
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import { ERC165Checker } from "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+import { ILicenseRegistryV2 } from "../interfaces/registries/ILicenseRegistryV2.sol";
+import { ILicensingModule } from "../interfaces/modules/licensing/ILicensingModule.sol";
+import { IDisputeModule } from "../interfaces/modules/dispute/IDisputeModule.sol";
+import { Errors } from "../lib/Errors.sol";
+import { Licensing } from "../lib/Licensing.sol";
+import { GovernableUpgradeable } from "../governance/GovernableUpgradeable.sol";
+import { ILicenseTemplate } from "contracts/interfaces/modules/licensing/ILicenseTemplate.sol";
+
+/// @title LicenseRegistry aka LNFT
+/// @notice Registry of License NFTs, which represent licenses granted by IP ID licensors to create derivative IPs.
+contract LicenseRegistryV2 is ILicenseRegistryV2, GovernableUpgradeable, UUPSUpgradeable {
```

Comment:
V2 only reserved for upgrades to deployed contract

Reply:
will remove the V2 before merging.